ci/remote-watch.py

```diff
@@ -0,0 +1,247 @@
+#!/usr/bin/env python
+
+"""
+This command must be run in a git repository.
+
+It watches the remote branch for changes, killing the given PID when it detects
+that the remote branch no longer points to the local commit.
+
+If the commit message contains a line saying "CI_KEEP_ALIVE", then killing
+will not occur until the branch is deleted from the remote.
+
+If no PID is given, then the entire process group of this process is killed.
+"""
+
+# Prefer to keep this file Python 2-compatible so that it can easily run early
+# in the CI process on any system.
+
+import errno
+import logging
+import os
+import re
+import signal
+import subprocess
+import sys
+import time
+
+from timeit import default_timer  # monotonic timer
+
+logger = logging.getLogger(__name__)
+
+
+def git(*args, **kwargs):
+    capture = kwargs.pop("capture", True)
+    assert len(kwargs) == 0, "Unexpected kwargs: {}".format(kwargs)
+    cmdline = ["git"] + list(args)
+    if capture:
+        result = subprocess.check_output(cmdline).decode("utf-8").rstrip()
+    else:
+        result = subprocess.check_call(cmdline)
+    return result
+
+
+def get_current_ci():
+    result = None
+    if "GITHUB_WORKFLOW" in os.environ:
+        result = "GitHub"
+    elif "TRAVIS" in os.environ:
+        result = "Travis"
+    return result
```

Comment:
Make GITHUB and TRAVIS constants for more readable comparison

```suggestion
    if "GITHUB_WORKFLOW" in os.environ:
        return GITHUB
    elif "TRAVIS" in os.environ:
        return TRAVIS
```